include/engine/routing_algorithms/map_matching.hpp

```diff
@@ -370,12 +371,8 @@ class MapMatching final : public BasicRoutingInterface<DataFacadeT, MapMatching<
             std::deque<std::pair<std::size_t, std::size_t>> reconstructed_indices;
             while (parent_timestamp_index > sub_matching_begin)
             {
-                if (model.breakage[parent_timestamp_index])
```

Comment:
It can lead to infinite loop. We don't change `parent_timestamp_index` or `sub_matching_begin` here, but continue to the while statement.